sql/src/main/java/io/crate/operation/reference/sys/check/cluster/LuceneVersionChecks.java

```diff
@@ -39,4 +45,31 @@ static boolean isUpgradeRequired(@Nullable String versionStr) {
             throw new IllegalArgumentException("'" + versionStr + "' is not a valid Lucene version");
         }
     }
+
+    // We need to check for the version that the index was created since the lucene segments might
+    // be automatically upgraded to the latest version (can happened when data was in the translog) so
+    // minimumCompatVersion cannot be used to indicate that a table recreation is needed.
+    static boolean isRecreationRequired(IndexMetaData indexMetaData) {
```

Comment:
`isRecreationRequired` and `tablesNeedRecreation` don't do any Lucene version checking so maybe they should live in `TablesNeedRecreationSysCheck`?